### PR TITLE
[hist] Implement axis slicing

### PR DESCRIPTION
**Pull request by hahnjo**
These will be used to implement slicing operations on histograms.

### CODE REVIEW

File: hist/histv7/inc/ROOT/RRegularAxis.hxx [hunk 1]
```diff
@@ -83,6 +83,24 @@ public:
    double GetHigh() const { return fHigh; }
    bool HasFlowBins() const { return fEnableFlowBins; }
 
+   /// Compute the low edge of a bin.
+   ///
+   /// \param[in] bin the index, must be \f$< fNNormalBins\f$
+   double ComputeLowEdge(std::uint64_t bin) const
+   {
+      assert(bin < fNNormalBins);
```
Comment (jblomer) [inline hist/histv7/inc/ROOT/RRegularAxis.hxx]: Being a public interface, should this throw instead?

File: hist/histv7/inc/ROOT/RRegularAxis.hxx [hunk 2]
```diff
@@ -212,6 +213,59 @@ public:
...
+            begin = range.GetBegin().GetIndex();
+            // Only compute a new lower end of the axis interval if needed.
+            if (begin > 0) {
+               low = ComputeLowEdge(begin);
+            }
+         }
+         if (range.GetEnd().IsNormal()) {
+            end = range.GetEnd().GetIndex();
+            // Only compute a new upper end of the axis interval if needed, to avoid floating-point inaccuracies.
+            if (end < nNormalBins) {
```
Comment (jblomer) [inline hist/histv7/inc/ROOT/RRegularAxis.hxx:253]: Not `< (nNormalBins - 1)`?

**Review by jblomer — APPROVED** [review]
LGTM

File: hist/histv7/inc/ROOT/RRegularAxis.hxx (hunk 2 above)
Comment (hahnjo) [reply hist/histv7/inc/ROOT/RRegularAxis.hxx:253]: The condition should be correct because `end` is exclusive. But I was confused for a minute as well because the tests didn't exercise the exact corner case. I'll strengthen them.
